scipy/stats/tests/test_discrete_basic.py

```diff
@@ -579,3 +580,44 @@ def _pmf(self, i):
     rv = rv_exponential(a=0.0, b=float('inf'))
     rvs = rv.rvs(random_state=42)  # should not crash due to integer input to `_pmf`
     assert_allclose(rvs, 0)
+
+
+def test_gh18919_ppf_array_args():
+    # gh-18919 reported incorrect results for ppf and isf of discrete distributions when
+    # arguments were arrays and first argument (`q`) had elements at the boundaries of
+    # the support.
+    q = [[0.5, 1.0, 0.5],
+         [1.0, 0.5, 1.0],
+         [0.5, 1.0, 0.5]]
+
+    n = [[45, 46, 47],
+         [48, 49, 50],
+         [51, 52, 53]]
+
+    p = 0.5
+
+    ref = _ufuncs._binom_ppf(q, n, p)
+    res = stats.binom.ppf(q, n, p)
+    np.testing.assert_allclose(res, ref)
+
+def test_gh18919_ppf_isf_array_args2():
+    # a more general version of the test above. Requires that arguments are broadcasted
```

Comment:
Nitpick: could we perform this test for at least one other distribution to check that it is fixed in throughout? I do not mean to add another generic test for the old infra, just spot checks here for more than just `binom`.